### PR TITLE
[SYCL][Graph]Fix and add E2E tests for using local accessors in graphs

### PR DESCRIPTION
**Pull request by Bensuo**
- Update UR tag for fix for updating local accessors on CUDA/HIP
- Add e2e tests covering local accessor usage

### CODE REVIEW

File: sycl/test-e2e/Graph/Inputs/whole_update_local_acc.cpp [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+#include "../graph_common.hpp"
+
+using T = int;
+
+auto add_graph_node(
+    exp_ext::command_graph<exp_ext::graph_state::modifiable> &Graph,
+    queue &Queue, size_t Size, size_t LocalSize, T *Ptr) {
+  return add_node(Graph, Queue, [&](handler &CGH) {
+    local_accessor<T, 1> localMem(LocalSize, CGH);
```
Comment (EwanC) [inline sycl/test-e2e/Graph/Inputs/whole_update_local_acc.cpp:12]:
```suggestion
    local_accessor<T, 1> LocalMem(LocalSize, CGH);
```

**Review by EwanC — COMMENTED** [review]
LGTM in general, left some minor comments. Main questions is if we need any new spec wording around local memory update?

**Review by EwanC — APPROVED** [review]

**Comment by callumfare** [comment]
@Bensuo @EwanC The UR PR has merged now. I've updated the UR tag but could you rebase/merge in the latest sycl branch?

**Comment by EwanC** [comment]
@callumfare Should be good now

**Review by callumfare — APPROVED** [review]

**Comment by EwanC** [comment]
@intel/llvm-gatekeepers This is good to go, thanks

**Pull request merged**